src/lib/leagues/masterTasks.ts

```diff
@@ -0,0 +1,1065 @@
+import { getTameSpecies } from '../tames';
+import { calcTotalLevel } from '../util';
+import resolveItems from '../util/resolveItems';
+import { Task } from './leagues';
+
+export const masterTasks: Task[] = [
+	{
+		id: 4000,
+		name: 'Build a Demonic Throne',
+		has: async ({ poh }) => {
+			return poh.throne === 13_671;
+		}
+	},
+	{
+		id: 4002,
+		name: 'Receive atleast 1000 Agility Arena tickets',
+		has: async ({ cl }) => {
+			return cl.amount('Agility arena ticket') >= 1000;
+		}
+	},
+	{
+		id: 4004,
+		name: 'Receive all Justiciar items',
+		has: async ({ cl }) => {
+			return resolveItems(['Justiciar faceguard', 'Justiciar chestguard', 'Justiciar legguards']).every(gs =>
+				cl.has(gs)
+			);
+		}
+	},
+	{
+		id: 4005,
+		name: 'Receive all Inquisitor armour pieces',
+		has: async ({ cl }) => {
+			return resolveItems(["Inquisitor's great helm", "Inquisitor's hauberk", "Inquisitor's plateskirt"]).every(
+				gs => cl.has(gs)
+			);
+		}
+	},
+	{
+		id: 4006,
+		name: 'Receive all Chambers of Xeric Weapons',
+		has: async ({ cl }) => {
+			return resolveItems([
+				'Elder maul',
+				'Twisted bow',
+				'Dragon claws',
+				'Dragon hunter crossbow',
+				'Kodai insignia'
+			]).every(gs => cl.has(gs));
+		}
+	},
+	{
+		id: 4007,
+		name: 'Receive all Chambers of Xeric Armor',
+		has: async ({ cl }) => {
+			return resolveItems([
+				'Ancestral hat',
+				'Ancestral robe top',
+				'Ancestral robe bottom',
+				"Dinh's bulwark"
+			]).every(gs => cl.has(gs));
+		}
+	},
+	{
+		id: 4008,
+		name: 'Receive all Chambers of Xeric items',
+		has: async ({ cl }) => {
+			return chambersOfXericCL.every(gs => cl.has(gs));
+		}
+	},
+	{
+		id: 4009,
+		name: 'Create every god book',
+		has: async ({ cl }) => {
+			return resolveItems([
+				'Book of balance',
+				'Book of darkness',
+				'Book of law',
+				'Book of war',
+				'Holy book',
+				'Unholy book'
+			]).every(gs => cl.has(gs));
+		}
+	},
+	{
+		id: 4010,
+		name: 'Create every pair of Cerberus boots',
+		has: async ({ cl }) => {
+			return resolveItems(['Primordial boots', 'Pegasian boots', 'Eternal boots']).every(gs => cl.has(gs));
+		}
+	},
+	{
+		id: 4011,
+		name: 'Receive every Revenant weapon',
+		has: async ({ cl }) => {
+			return resolveItems(["Craw's bow (u)", "Thammaron's sceptre (u)", "Viggora's chainmace (u)"]).every(gs =>
+				cl.has(gs)
+			);
+		}
+	},
+	{
+		id: 4012,
+		name: 'Achieve base level 110 stats',
+		has: async ({ skillsLevels }) => {
+			return Object.values(skillsLevels).every(i => i >= 110);
+		}
+	},
+	{
+		id: 4013,
+		name: 'Create every draconic shield',
+		has: async ({ cl }) => {
+			return resolveItems(['Dragonfire ward', 'Dragonfire shield', 'Ancient wyvern shield']).every(gs =>
+				cl.has(gs)
+			);
+		}
+	},
+	{
+		id: 4014,
+		name: 'Create every nightmare staff',
+		has: async ({ cl }) => {
+			return resolveItems([
+				'Volatile nightmare staff',
+				'Harmonised nightmare staff',
+				'Eldritch nightmare staff'
```

Comment:
This can be cheesed by taking apart a staff and re-combining it. It can also be shared by players this way.

I suggest you check CL for the Orbs as well.